rfc/0054-sdk3-management-apis.md

```diff
@@ -1494,6 +1530,190 @@ Map where top level keys are dataverse names, and values are a map of dataset na
+  * The URI is:
+    * POST http://localhost:8095/analytics/link
+
+
+## ReplaceLink
+
+Replaces an existing link.
+
+### Signature
+
+```
+void ReplaceLink(AnalyticsLink link, [options])
+```
+
+### Parameters
+
+* Required:
+
+  * `link`: `AnalyticsLink` - the link to be replaced.
+
+* Optional:
+
+  * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
+
+### Returns
+
+Nothing
+
+### Throws
+
+* `DataverseNotFoundException`
+
+* `AnalyticsLinkNotFoundException`
+
+* `InvalidArgumentsException`
+
+### URI
+
+Data is sent as `"application/x-www-form-urlencoded"`.
+
+* The link's `dataverse` can contain one or more `/`.
+* The path to use depends on the `dataverse`.
+* If `dataverse` contains a `/` then the name is url encoded to escape any `/` within the name and URI is:
+  * PUT http://localhost:8095/analytics/link/<dataverse>/<linkname>
+* If `dataverse` does not contain a `/` then the `dataverse` and `name` are included within the payload of the request (as `dataverse` and `name`)
+  * The URI is:
+    * POST http://localhost:8095/analytics/link
+
+## DropLink
+
+Drops an existing link from a scope.
+
+### Signature
+
+```
+void DropLink(string linkName, string scopeName, [options])
+```
+
+### Parameters
+
+* Required:
+
+  * `linkName`: `string` - the link to be removed.
+  * `scopeName`: `string` - the scope in which the link belongs, in the format `bucket/scope`.
+
+* Optional:
+
+  * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
+
+### Returns
+
+Nothing
+
+### Throws
+
+* `DataverseNotFoundException`
+
+* `AnalyticsLinkNotFoundException`
+
+* `InvalidArgumentsException`
+
+### URI
+
+* The link's `dataverse` can contain one or more `/`.
+* The path to use depends on the `dataverse`.
+* If `dataverse` contains a `/` then the name is url encoded to escape any `/` within the name and URI is:
+  * DELETE http://localhost:8095/analytics/link/<dataverse>/<linkname>
+* If `dataverse` does not contain a `/` then the `dataverse` and `name` are included within the payload of the request (as `dataverse` and `name`)
+  * The URI is:
+    * POST http://localhost:8095/analytics/link
+
+## GetAllLinks
+
+Gets all existing links.
+
+### Signature
+
+```
+List<AnalyticsLink> GetAllLinks([options])
+```
+
+### Parameters
+
+* Required:
+
+None
+
+* Optional:
+
+  * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
+  * `Dataverse` (string) - the name of the dataverse to restrict links to.
+  * `Name` (string) - the name of the link to fetch.
+  * `LinkType` (AnalyticsLinkType) - the type of links to restrict returned links to.
+  
+Note: If `Name` is set then `Dataverse` must be set, otherwise an `InvalidArgumentsException` must be thrown.
```

Comment:
If `name` is set but not `dataverse`, should we fall back to sending `name` as a query parameter? (It seems to work, just wondering if there's a reason not to do it.)